common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java

```diff
@@ -205,47 +247,81 @@ public void render(PoseStack poseStack, float partialTicks, Window window) {
                 }
             }
 
-            // TODO compass icon
+            // Compass icon
+            Optional<WaypointPoi> compassOpt = CompassModel.getCompassWaypoint();
 
-            // cursor
-            if (!followPlayerRotation) {
-                poseStack.pushPose();
-                RenderUtils.rotatePose(
-                        poseStack, centerX, centerZ, 180 + McUtils.player().getYRot());
-            }
+            if (compassOpt.isEmpty()) return;
 
-            MapRenderer.renderCursor(
-                    poseStack, centerX, centerZ, this.pointerScale, this.pointerColor, this.pointerType);
+            WaypointPoi compass = compassOpt.get();
 
-            if (!followPlayerRotation) {
-                poseStack.popPose();
+            Pair<Float, Float> compassLocation = getRenderLocation(
+                    compass, playerX, playerZ, centerX, centerZ, sinRotationRadians, cosRotationRadians);
+            float compassRenderX = compassLocation.a();
+            float compassRenderZ = compassLocation.b();
+
+            final float compassSize = Math.max(compass.getWidth(), compass.getHeight()) * 0.8f * poiScale;
+
```

Comment:
0.8 is just scaling down right? Is the texture too big?

Reply:
Copied from legacy